lib/src/repo.rs

```diff
@@ -383,17 +383,21 @@ impl StoreFactories {
     }
 
     pub fn load_backend(&self, store_path: &Path) -> Box<dyn Backend> {
-        // TODO: Change the 'backend' file to 'type', for consistency with other stores
-        let backend_type = match fs::read_to_string(store_path.join("backend")) {
+        // For compatibility with existing repos. TODO: Delete in 0.8+.
```

Comment:
To not prevent us from releasing often, I think time-triggered removals are best in general. As soon as `jj` enters a Linux distribution, we should allow people from getting from one distribution version to the other (think Ubuntu LTS). A common format, such as "something AT DATE: yyyy-mm-dd" could even be checked by the CI.

Reply:
Moreover, I think that it would be best to store a "repo version" somewhere, so that we make sure that `jj` does not get downgraded to a previous incompatible version (which is easy for repos stored on network file systems). This way migration paths could be well identified and cleaner.

Reply:
Yes, we need some mechanism for making too old and too new versions error out instead of crashing. I'm not sure how that should look yet.  Maybe a simple version number is enough. Mercurial has a `.hg/requires` file that contains a list of features any client must understand. Maybe there's less need for that since we don't support extensions in the same way. Good point about Linux distributions.

Anyway, I'll merge this now since figuring that out is a larger question.